sdk/Lusid.Sdk/Utilities/LusidApiFactoryBuilder.cs

```diff
@@ -20,31 +14,20 @@ public static ILusidApiFactory Build(string apiSecretsFilename)
             return new LusidApiFactory(apiConfig);
         }
 
-
         /// <summary>
         /// Create an ILusidApiFactory using the specified Url and Token Provider
         /// </summary>
         public static ILusidApiFactory Build(string url, ITokenProvider tokenProvider)
         {
-            lock (Lock)
+            // TokenProviderConfiguration.ApiClient is the client used by LusidApiFactory and is 
+            // NOT thread-safe, so there needs to be a separate instance for each instance of LusidApiFactory.
+            // Do NOT cache the LusidApiFactory instances (DEV-6922)
+            var config = new TokenProviderConfiguration(tokenProvider)
             {
-                var threadId = Thread.CurrentThread.ManagedThreadId;
-
-                if (!ThreadFactories.TryGetValue(threadId, out var factory))
-                {
-                    // TokenProviderConfiguration.ApiClient is the client used by LusidApiFactory and is 
-                    // not threadsafe, so there needs to be a separate instance for each instance of LusidApiFactory
-                    var config = new TokenProviderConfiguration(tokenProvider)
-                    {
-                        BasePath = url
-                    };
-
-                    factory = new LusidApiFactory(config);
-                    ThreadFactories[threadId] = factory;
-                }
+                BasePath = url
+            };
 
-                return factory;
-            }
+            return new LusidApiFactory(config);
```

Comment:
No need to cache these, because:
(a) you DO want a new instance each time, because the factory isn't thread safe, because the contained API instances are not thread safe
(b) these objects are (now) inexpensive to create